synadm/api.py

```diff
@@ -1139,3 +1141,58 @@ def user_shadow_ban(self, user_id, unban):
         else:
             method = "post"
         return self.query(method, f"v1/users/{user_id}/shadow_ban")
+
+    def notice_send(self, receivers, content_plain, content_html, batch, paginate):
+        """ Send server notices.
+
+        Args:
+            receivers: Target(s) of the notice. Either localpart or regular 
+                expression matching localparts.
+            content_plain: Unformatted text of the notice.
+            content_html: HTML-formatted text of the notice.
+        """
+        data = {
+                "user_id": "",
+                "content": {
+                        "msgtype": "m.text",
+                        "body": content_plain,
+                        "format": "org.matrix.custom.html",
+                        "formatted_body": content_html
+                }
+        }
+        def ask_confirmation(users):
+            if batch:
+                return True
+            prompt = "Recipients (list may be incomplete):\n"
+            ctr = 0
+            for user in users:
+                if not re.search(receivers, user) is None:
+                    prompt = prompt + " - " + user + "\n"
+                    ctr = ctr + 1
+                    if ctr == 10:
+                        prompt = prompt + " - ..."
+                        break
+            prompt = prompt + "\nUnformatted message:\n---\n" + content_plain + "\n---\nFormatted message:\n---\n" + content_html + "\n---\nSend now?"
+            return click.confirm(prompt)
+
+        # Only a single user ID was supplied as receiver
```

Comment:
in api.py you will find some helper method around usernames. synadm now supports automagically determining if a username is only localpart or is a full mxid. use the eisting helper please.

Reply:
Please also make sure that you do not exceed the maximum line length of 79 characters.